src/libopensc/card-isoApplet.c

```diff
@@ -210,22 +215,33 @@ isoApplet_init(sc_card_t *card)
 	card->cla = 0x00;
 
 	/* Obtain applet version and specific features */
-	if (0 > isoApplet_select_applet(card, isoApplet_aid, ISOAPPLET_AID_LEN, rbuf, &rlen)) {
-		free(card->drv_data);
-		card->drv_data = NULL;
-		LOG_TEST_RET(card->ctx, SC_ERROR_INVALID_CARD, "Error obtaining applet version.");
+	r = isoApplet_get_info(card, drvdata);
+	LOG_TEST_GOTO_ERR(card->ctx, r, "Error obtaining information about applet.");
+
+
+	unsigned int major_version = drvdata->isoapplet_version & 0xFF00;
```

Comment:
I would probably advocate to move this variable declaration before any any other statements.

Reply:
Done